main.py

```
@@ -113,5 +113,160 @@ async def list_esp_serial_ports() -> (str, str):
+    
+    returncode, stdout, stderr = await run_command_async(
+        f"bash -c 'source {export_script} && idf.py size-components'"
+    )
+    
+    open('mcp-size-components.log', 'w+').write(str((stdout, stderr)))
+    logging.warning(f"size-components result {stdout} {stderr}")
+    return stdout, stderr
+
+
+@mcp.tool()
+async def erase_esp_flash(project_path: str, port: str = None) -> (str, str):
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)